models/migrations/v105.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func prependRefsHeadsToIssueRefs(x *xorm.Engine) error {
+	type Issue struct {
+		ID  int64
+		Ref string
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	const batchSize = 100
+	count := 0
+	for start := 0; ; start += batchSize {
+		issues := make([]*Issue, 0, batchSize)
+		if err := sess.Limit(batchSize, start).Asc("id").Find(&issues); err != nil {
+			return err
+		}
+
+		if len(issues) == 0 {
+			break
+		}
+
+		for _, issue := range issues {
```

Comment:
No need to update one by one. This can be done is single SQL command:
```sql
update `issue` set `ref` = 'refs/heads/' || `ref` where `ref` is not null and `ref` != '';
```

Reply:
Will this be compatible with all database engines? I've looked at a bunch of other migrations for reference (v82 and v85 from the top of my head) and they did it the exact same way

Reply:
Needs to be tested but this should work imho on all DB types:
```go
func prependRefsHeadsToIssueRefs(x *xorm.Engine) error {
	_, err := x.Exec("UPDATE `issue` SET `ref` = 'refs/heads/' || `ref` WHERE `ref` is not null AND `ref` != ''")
	return err
}
```

Reply:
PostgreSQL does not support using backticks for table/column name quoting, so I left them out. They aren't keywords anyway

Reply:
After some more research I found out that `ref` is in fact reserved in more modern SQL standards. 

Also, escaping reserved words normally is done with double quotes, but MySQL is the special one using backticks, with no support for double quotes unless you start it in ANSI mode.

Reply:
@SijmenSchoon there are a couple of migrations that are "engine aware" (e.g. the drop column function). I guess you could resort to that if necessary.

Reply:
I opened [an xorm issue](https://gitea.com/xorm/core/issues/62) for future cases, and I just made it engine aware. Let's see if this works